### ISSUE TITLE
"containerEngine.imageAllowList.patterns" entry is not generated while updating options.go

### ISSUE THREAD

**Issue by s0nea** [rancher-sandbox/rancher-desktop#3995]
### Actual Behavior

I'm not sure if the current behavior is on purpose. I'm creating this issue just in case it's not. Please just close the issue, if everything works as expected.

I noticed the message
`Not generating a CLI entry for property containerEngine.imageAllowList.patterns: arrays not supported.`
while running
`node scripts/ts-wrapper.js scripts/generateCliCode.ts pkg/rancher-desktop/assets/specs/command-api.yaml src/go/rdctl/cmd/options.go`
to update `options.go`.
I assume this message is related to `patterns: [] as Array<string>` in https://github.com/rancher-sandbox/rancher-desktop/blob/main/pkg/rancher-desktop/config/settings.ts#L60 .

### Steps to Reproduce

- Get the latest main version
- Run `node scripts/ts-wrapper.js scripts/generateCliCode.ts pkg/rancher-desktop/assets/specs/command-api.yaml src/go/rdctl/cmd/options.go`

### Result

The CLI entry is not generated:

```
╰─$ node scripts/ts-wrapper.js scripts/generateCliCode.ts pkg/rancher-desktop/assets/specs/command-api.yaml src/go/rdctl/cmd/options.go                                                                                                  1 ↵
Generating src/go/rdctl/cmd/options.go...
Not generating a CLI entry for property containerEngine.imageAllowList.patterns: arrays not supported.
Skipping WSL.integrations: not settable from the command-line.
Skipping diagnostics.mutedChecks: not settable from the command-line.
```

### Expected Behavior

The CLI entry is generated?

### Rancher Desktop Version

latest / main branch

### Rancher Desktop K8s Version

1.25.6

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

Other Linux

### Operating System / Build Version

openSUSE Leap 15.3

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

rpm

**Comment by ericpromislow** [comment]
I can't find an issue that has us implementing support for arrays and hashes in `rdctl start` and `rdctl set`, but they depend on this issue getting fixed. But currently this is expected behaviour.

**Comment by jandubois** [comment]
> I can't find an issue that has us implementing support for arrays and hashes

I don't think we have one yet. I still have this on my list to figure out what kind of syntax we would want to use. There are so many operations you may want to perform:

* add an entry
* remove an entry
* replace an entry
* clear the whole list
* replace the whole list

And in this case order does not really matter, otherwise you might want to allow something like "add this entry at index N, shifting everything below on entry down", or prepend at the beginning, append at the end.

For the programmatic API we don't need this: the client can fetch the whole list; manipulate it however it wants locally, and then send a change request with the updated list. But that is not convenient for a scripting API.

On the other hand, this is a niche use case, so hard to justify spending a lot of effort on it. As a fallback you can always manipulate it via `rdctl api`, but that works only if the app is already running.
